verifier/src/report_body.rs

```diff
@@ -274,6 +274,47 @@ impl<T: Accessor<MrEnclave>> Verifier<T> for MrEnclaveVerifier {
     }
 }
 
+/// Verify the [`MrSigner`] is as expected.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct MrSignerVerifier {
+    expected_mr_signer: MrSigner,
+}
+
+impl MrSignerVerifier {
+    /// Create a new [`MrSignerVerifier`] instance.
+    ///
+    /// # Arguments:
+    /// * `expected_mr_signer` - The expected MRSIGNER.
+    pub fn new(expected_mr_signer: MrSigner) -> Self {
+        Self { expected_mr_signer }
+    }
+}
+
+impl Accessor<MrSigner> for ReportBody {
+    fn get(&self) -> MrSigner {
+        let Measurement::MrSigner(mr_signer) = self.mr_signer() else {
+            panic!("`mr_signer()` should return a Measurement::MrSigner");
+        };
+        mr_signer
+    }
```

Comment:
Similar to #23
This is here until, https://github.com/mobilecoinfoundation/sgx/issues/286.
Added as a task to #7